ic10emu/src/vm/instructions/traits.rs

```diff
@@ -0,0 +1,4615 @@
+use crate::vm::object::traits::IntegratedCircuit;
+use crate::vm::instructions::enums::InstructionOp;
+pub trait AbsInstruction: IntegratedCircuit {
+    ///abs r? a(r?|num)
+    fn execute_abs(
+        &mut self,
+        r: &crate::vm::instructions::operands::Operand,
+        a: &crate::vm::instructions::operands::Operand,
+    ) -> Result<(), crate::errors::ICError> {
```

Comment:
this function uses Operand because it and it's call is generated. it performs the transformation to `InstOperand` and calls `execute_inner` which is hand written.

Reply:
The reason to include it would be to save on bytes and cognitive load, there's no reason to always qualify here

Reply:
.. I think my brain may have skipped when I first read that. for some reason I was thinking you were suggesting that converting the function to use InstOperand. but yeah, your correct I think.